agent/src/agent/modules/field.py

```diff
@@ -73,4 +73,4 @@ def build_fields(fields_conf: dict) -> list[Field]:
 
 
 def extract_fields(fields: list[Field], data: dict) -> dict:
-    return {field_.get_name(): field_.extract_from(data) for field_ in fields}
+    return {field_.get_name(): field_.extract_from(data) for field_ in fields if field_.extract_from(data) != ''}
```

Comment:
I don't think this is a good idea, first, transformations are applied two times here for every field which increases the load, second, it compares to a string, but could it be None?